From b442b6fe3d571cbc4fb943322fe25195df4758a7 Mon Sep 17 00:00:00 2001
From: Marcel Ziswiler <marcel.ziswiler@toradex.com>
Date: Mon, 16 Dec 2019 10:48:19 +0100
Subject: [PATCH 07/18] arm: dts: imx6dl/q/ull/7-apalis/colibri-eval/eval-v3:
 vesa vga display timings

Use proper VESA VGA display timings by default.

Signed-off-by: Marcel Ziswiler <marcel.ziswiler@toradex.com>
---
 arch/arm/boot/dts/imx6dl-colibri-eval-v3.dts  | 14 +++++++------
 arch/arm/boot/dts/imx6q-apalis-eval.dts       | 14 +++++++------
 .../arm/boot/dts/imx6ull-colibri-eval-v3.dtsi | 21 +++++++++++++++++--
 arch/arm/boot/dts/imx7-colibri-eval-v3.dtsi   | 14 +++++++------
 4 files changed, 43 insertions(+), 20 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dl-colibri-eval-v3.dts b/arch/arm/boot/dts/imx6dl-colibri-eval-v3.dts
index 70e870385a09..0e27f180913b 100644
--- a/arch/arm/boot/dts/imx6dl-colibri-eval-v3.dts
+++ b/arch/arm/boot/dts/imx6dl-colibri-eval-v3.dts
@@ -129,15 +129,17 @@
 		data-mapping = "bgr666";
 
 		panel-timing {
+			/* Default VESA VGA display timings */
 			clock-frequency = <25175000>;
 			hactive = <640>;
-			vactive = <480>;
-			hsync-len = <30>;
+			hback-porch = <48>;
 			hfront-porch = <16>;
-			hback-porch = <114>;
-			vsync-len = <3>;
-			vfront-porch = <10>;
-			vback-porch = <32>;
+			hsync-len = <96>;
+			vactive = <480>;
+			vback-porch = <31>;
+			vfront-porch = <11>;
+			vsync-len = <2>;
+			pixelclk-active = <0>;
 		};
 
 		port {
diff --git a/arch/arm/boot/dts/imx6q-apalis-eval.dts b/arch/arm/boot/dts/imx6q-apalis-eval.dts
index 9ba06b091790..550409bc0825 100644
--- a/arch/arm/boot/dts/imx6q-apalis-eval.dts
+++ b/arch/arm/boot/dts/imx6q-apalis-eval.dts
@@ -103,15 +103,17 @@
 		data-mapping = "rgb565";
 
 		panel-timing {
+			/* Default VESA VGA display timings */
 			clock-frequency = <25175000>;
 			hactive = <640>;
-			vactive = <480>;
-			hsync-len = <30>;
+			hback-porch = <48>;
 			hfront-porch = <16>;
-			hback-porch = <114>;
-			vsync-len = <3>;
-			vfront-porch = <10>;
-			vback-porch = <32>;
+			hsync-len = <96>;
+			vactive = <480>;
+			vback-porch = <31>;
+			vfront-porch = <11>;
+			vsync-len = <2>;
+			pixelclk-active = <0>;
 		};
 
 		port {
diff --git a/arch/arm/boot/dts/imx6ull-colibri-eval-v3.dtsi b/arch/arm/boot/dts/imx6ull-colibri-eval-v3.dtsi
index e27abcc0a7fe..24af2b426337 100644
--- a/arch/arm/boot/dts/imx6ull-colibri-eval-v3.dtsi
+++ b/arch/arm/boot/dts/imx6ull-colibri-eval-v3.dtsi
@@ -15,10 +15,27 @@
 		clock-frequency = <16000000>;
 	};
 
-	panel: panel {
-		compatible = "edt,et057090dhu";
+	panel_dpi: panel-dpi {
+		compatible = "panel-dpi";
 		backlight = <&bl>;
+		data-mapping = "bgr666";
+		height-mm = <86>;
 		power-supply = <&reg_3v3>;
+		width-mm = <115>;
+
+		panel-timing {
+			/* Default VESA VGA display timings */
+			clock-frequency = <25175000>;
+			hactive = <640>;
+			hback-porch = <48>;
+			hfront-porch = <16>;
+			hsync-len = <96>;
+			vactive = <480>;
+			vback-porch = <31>;
+			vfront-porch = <11>;
+			vsync-len = <2>;
+			pixelclk-active = <0>;
+		};
 
 		port {
 			panel_in: endpoint {
diff --git a/arch/arm/boot/dts/imx7-colibri-eval-v3.dtsi b/arch/arm/boot/dts/imx7-colibri-eval-v3.dtsi
index 1fc754cf2887..77a1066c37c0 100644
--- a/arch/arm/boot/dts/imx7-colibri-eval-v3.dtsi
+++ b/arch/arm/boot/dts/imx7-colibri-eval-v3.dtsi
@@ -82,15 +82,17 @@
 		data-mapping = "bgr666";
 
 		panel-timing {
+			/* Default VESA VGA display timings */
 			clock-frequency = <25175000>;
 			hactive = <640>;
-			vactive = <480>;
-			hsync-len = <30>;
+			hback-porch = <48>;
 			hfront-porch = <16>;
-			hback-porch = <114>;
-			vsync-len = <3>;
-			vfront-porch = <10>;
-			vback-porch = <32>;
+			hsync-len = <96>;
+			vactive = <480>;
+			vback-porch = <31>;
+			vfront-porch = <11>;
+			vsync-len = <2>;
+			pixelclk-active = <0>;
 		};
 
 		port {
-- 
2.25.1

